caffe2/core/nomnigraph/include/nomnigraph/Graph/Algorithms.h

```diff
@@ -119,7 +119,8 @@ Graph<typename G::NodeRef, int> dominatorTree(
   nextPass.insert(source);
 
   while (nextPass.size()) {
-    for (auto parent : nextPass) {
+    for (auto parent_iter = nextPass.begin(); parent_iter != nextPass.end(); ) {
```

Comment:
A safer way to do it may be through 2 pass. First pass to insert and push the node to be deleted in a vector. And second pass to iterate through the vector to erase. Something like 
```
vector<typename G::NodeRef> to_delete;
for (auto parent: nextPass) {
  to_delete.push_back(parent);
  ...
}

for (const auto key: to_delete) {
  nextPass.erase(key);
}
```

A bit more costly but cleaner and safer.